dace/codegen/targets/xilinx.py

```diff
@@ -515,16 +522,21 @@ def generate_module(self, sdfg, state, name, subgraph, parameters,
         added = set()
 
         parameters = list(sorted(parameters, key=lambda t: t[1]))
-        arrays = [
-            p for p in parameters if not isinstance(p[2], dace.data.Scalar)
-        ]
+        arrays = dtypes.deduplicate([p for p in parameters
+                                     if not isinstance(p[2], dace.data.Scalar)])
         scalars = [p for p in parameters if isinstance(p[2], dace.data.Scalar)]
-        scalars += ((False, k, v) for k, v in symbol_parameters.items())
+        scalars += ((False, k, v, None) for k, v in symbol_parameters.items())
         scalars = dace.dtypes.deduplicate(sorted(scalars, key=lambda t: t[1]))
-        for is_output, pname, p in itertools.chain(parameters, scalars):
+        for is_output, pname, p, interface_id in itertools.chain(arrays, scalars):
```

Comment:
@definelicht note that I changed `parameters` to `arrays` here. It was probably wrong in the original code, since `arrays` was unused